.github/workflows/atlas-tests.yml

```diff
@@ -0,0 +1,134 @@
+## Github workflow to run atlas tests
+
+name: Tests::Atlas
+
+on:
+  workflow_call:
+
+defaults:
+  run:
+    shell: bash
+
+env:
+  BITCOIND_TEST: 1
+  RUST_BACKTRACE: full
+  SEGMENT_DOWNLOAD_TIMEOUT_MINS: 15
+
+concurrency:
+  group: atlas-tests-${{ github.workflow }}-${{ github.ref }}
+  # Only cancel in progress if this is for a PR
+  cancel-in-progress: ${{ github.event_name == 'pull_request' }}
+
+jobs:
+  # atlas integration tests with code coverage
+  atlas-tests:
+    name: Atlas Test
+    runs-on: ubuntu-latest
+    ## run a maximum of 2 concurrent tests from the test matrix
+    strategy:
+      fail-fast: false
+      max-parallel: 2
+      matrix:
+        test-name:
+          - tests::neon_integrations::atlas_integration_test
+          - tests::neon_integrations::atlas_stress_integration_test
+    steps:
+      ## Checkout the code
+      - name: Checkout the latest code
+        id: git_checkout
+        uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9 # v3.5.3
+
+      ## Install rust toolchain (llvm-tools-preview)
+      - name: Setup Rust Toolchain
+        id: setup_rust_toolchain
+        uses: actions-rust-lang/setup-rust-toolchain@f3c84ee10bf5a86e7a5d607d487bf17d57670965 # v1.5.0
+        with:
+          toolchain: stable
+          components: llvm-tools-preview
+          cache: false
+
+      ## Restore cargo cache data
+      - name: Restore Cargo Cache
```

Comment:
Clarifying - why do we need to do anything with the cargo cache?

Reply:
we don't - but having the cargo cache makes executing the tests much faster to start (since all the cargo dependencies are restored via that cache).

the caching is not foolproof though - i have seen the occasional instance where a cache fails to restore, resulting in any test relying on that cache also failing. it's rare, but it does happen and there's no clear reason as to why (the logs only show a download pausing at %x until the job times out).

Reply:
in lieu of using the cache, we'd have to install the cargo dependencies for each cargo job

Reply:
one important piece of info i forgot to add: the cargo cache actually has the binaries we need to run (`nextest`). keeping them in a cache improves the runtime of later jobs (but it's not technically required, we could always re-install and sacrifice a little time)